mirar/processors/photcal/zp_calculator/color_term_zp_calculator.py

```diff
@@ -0,0 +1,71 @@
+"""
+Module to calculate zero point by including a color term.
+"""
+import logging
+from typing import Callable
+
+import numpy as np
+from astropy.table import Table
+from scipy.optimize import curve_fit
+
+from mirar.data import Image
+from mirar.processors.photcal.zp_calculator.base_zp_calculator import (
+    BaseZeroPointCalculator,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def line_model(data, slope, intercept):
+    """linear model to hand to scipy curve_fit"""
+    return slope * data + intercept
+
+
+class ZPWithColorTermCalculator(BaseZeroPointCalculator):
+    """
+    Class to calculate zero point by including a color term. This models the data as
+
+    ref_mag - img_mag = ZP + C * (ref_color)
+
+    Attributes:
+        color_colnames_generator: function that takes an image as input and returns
+        a list containing two strings that are the column names of the reference catalog
+        to use for the color term. The first string is the bluer band, the second is the
+         redder band.
+    """
+
+    def __init__(
+        self,
+        color_colnames_generator: Callable[[Image], list[str, str]],
+    ):
+        self.color_colnames_generator = color_colnames_generator
+
+    def calculate_zeropoint(
+        self,
+        image: Image,
+        matched_ref_cat: Table,
+        matched_img_cat: Table,
+        colnames: list[str],
+    ) -> Image:
+        color_colnames = self.color_colnames_generator(image)
+        colors = matched_ref_cat[color_colnames[0]] - matched_ref_cat[color_colnames[1]]
+
+        for colname in colnames:
+            y = matched_img_cat[colname] - matched_ref_cat["magnitude"]
+            x = colors
+            y_err = np.sqrt(
+                matched_img_cat[colname.replace("MAG", "MAGERR")] ** 2
```

Comment:
when I try to run this, I get the error `  File "/Users/saarahhall/Desktop/research/mirar/mirar/processors/photcal/zp_calculator.py", line 161, in calculate_zeropoint
    matched_img_cat[colname.replace("MAG", "MAGERR")] ** 2
    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/saarahhall/miniconda3/envs/mirar/lib/python3.11/site-packages/astropy/table/table.py", line 2055, in __getitem__
    return self.columns[item]
           ~~~~~~~~~~~~^^^^^^
  File "/Users/saarahhall/miniconda3/envs/mirar/lib/python3.11/site-packages/astropy/table/table.py", line 264, in __getitem__
    return OrderedDict.__getitem__(self, item)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'MAGERRAPER_4.0' ` Is this an issue on my end or have you tested this function out?